pkg/controller/nodes/task/k8s/plugin_manager.go

```diff
@@ -109,8 +97,27 @@ type PluginManager struct {
 	kubeClient      pluginsCore.KubeClient
 	metrics         PluginMetrics
 	// Per namespace-resource
-	backOffController    *backoff.Controller
-	resourceLevelMonitor *ResourceLevelMonitor
+	backOffController             *backoff.Controller
+	resourceLevelMonitor          *ResourceLevelMonitor
+	overrideInjectOwnerReferences *bool
+	overrideInjectFinalizer       *bool
+}
+
+func (e *PluginManager) AddObjectMetadata(taskCtx pluginsCore.TaskExecutionMetadata, o client.Object, cfg *config.K8sPluginConfig) {
+	o.SetNamespace(taskCtx.GetNamespace())
+	o.SetAnnotations(utils.UnionMaps(cfg.DefaultAnnotations, o.GetAnnotations(), utils.CopyMap(taskCtx.GetAnnotations())))
+	o.SetLabels(utils.UnionMaps(o.GetLabels(), utils.CopyMap(taskCtx.GetLabels()), cfg.DefaultLabels))
+	o.SetName(taskCtx.GetTaskExecutionID().GetGeneratedName())
+
+	if e.overrideInjectOwnerReferences != nil && *e.overrideInjectOwnerReferences {
+		o.SetOwnerReferences([]metav1.OwnerReference{taskCtx.GetOwnerReference()})
+	}
+
+	overrideInjectFinalizer := e.overrideInjectFinalizer
+	if (overrideInjectFinalizer != nil && *overrideInjectFinalizer) || cfg.InjectFinalizer {
```

Comment:
this condition again, is a little confusing
we will also inject if 
cfg.InjectFinalizer is true?
OR
if the override sets it to true?

should it not be
overrideInjectFinalizer != nil && *overrideInjectFinalizer && cfg.InjectFinalizer